File: ObjcExample/TestProj/ViewController.m
//
//  ViewController.m
//  TestProj
//
//  Created by Nick Wang on 28/06/21.
//  Copyright © 2021 Nick Wang. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet NestedSpinnerView *spinner;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"Examples";
    _spinner.dataSource = @[ [[Entity alloc] init:@"Simple Spinner"], [[Entity alloc] init:@"Nested Spinner"] ];
    _spinner.anchorOffset = CGPointMake(0, _spinner.bounds.size.height);
    _spinner.selectionAction = ^(NSIndexPath * _Nonnull indexPath, NSString * _Nonnull value, id _Nullable userdata) {
        if ([userdata isKindOfClass:[Entity class]]) {
            Entity *entity = (Entity *)userdata;
            if ([entity.name isEqualToString:@"Simple Spinner"]) {
                [self performSegueWithIdentifier:@"identifierSimpleSpinner" sender:nil];
            } else if ([entity.name isEqualToString:@"Nested Spinner"]) {
                [self performSegueWithIdentifier:@"identifierNestedSpinner" sender:nil];
            }
        }
    };
}

@end

File: Example/NestedSpinner/ViewController.swift
//
//  ViewController.swift
//  NestedSpinner
//
//  Created by Nick on 8/06/21.
//  Copyright © 2021 Nick Wang. All rights reserved.
//

import UIKit
import NestedSpinner

class ViewController: UIViewController {
    
    @IBOutlet weak var spinner: NestedSpinnerView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Examples"
        spinner.dataSource = [Entity("Simple Spinner"), Entity("Nested Spinner")]
        spinner.anchorOffset = CGPoint(x: 0, y: spinner.bounds.size.height)
        spinner.selectionAction = { indexPath, value, userdata in
            if userdata is Entity {
                let entity = userdata as! Entity
                switch entity.name {
                case "Simple Spinner":
                    self.performSegue(withIdentifier: "identifierSimpleSpinner", sender: nil)
                    break
                case "Nested Spinner":
                    self.performSegue(withIdentifier: "identifierNestedSpinner", sender: nil)
                    break
                default:
                    break
                }
            }
        }
    }
    
}
